src/porepy/numerics/mixed_dim/dof_manager.py

```diff
@@ -198,42 +196,115 @@ def distribute_variable(
 
                     data[pp.STATE][var_name] = vals
                 else:
-                    # If no values exist, there is othing to add to
+                    # If no values exist, there is nothing to add to
                     data[pp.STATE] = {var_name: values[dof[bi] : dof[bi + 1]]}
 
-    def transform_dofs(
-        self, dofs: np.ndarray, var: Optional[list] = None
-    ) -> np.ndarray:
-        """Transforms dofs associated to full list of dofs to a restricted list of dofs."""
-
-        # TODO this procedure should only be performed once! One could consider storing
-        # the projection matrix.
-        # Double-check whether dofs are actually represented by var.
-        total_dofs: int = np.sum(self.full_dof)  # type: ignore
-        if var is not None:
-            is_var_dofs = np.zeros(total_dofs, dtype=bool)
-            if not isinstance(var, list):
-                var = [var]
-            for grid, variable in self.block_dof:
-                if variable in var:
-                    var_dof_ind = self.dof_ind(grid, variable)
-                    is_var_dofs[var_dof_ind] = True
-            assert all(is_var_dofs[dofs])
-
-        # Input dofs in the context of all dofs managed by DofManager
-        total_dofs: int = np.sum(self.full_dof)  # type: ignore
-        global_dofs = np.zeros(total_dofs)
-        global_dofs[dofs] = 1
-
-        # Projection matrix from space of all dofs to restricted list of var
-        num_dofs = self.num_dofs(var=var)  # type: ignore
-        projection = sps.coo_matrix(
-            (np.arange(num_dofs), (np.arange(num_dofs), dofs)),
-            shape=(num_dofs, total_dofs),
-        )
+    def distribute_restricted_variable(
```

Comment:
What does restricted mean? How is this different from distribute variable? Should be clear in function documentation.